query-engine/core/src/query_graph_builder/write/nested/connect_nested.rs

```diff
@@ -361,6 +356,28 @@ fn handle_one_to_one(
         ));
     }
 
+    if parent_is_create {
+        handle_one_to_one_parent_create(graph, parent_node, parent_relation_field, filter, child_model)
+    } else {
+        handle_one_to_one_parent_update(graph, parent_node, parent_relation_field, filter, child_model)
+    }
```

Comment:
The diff of the split is not super easy to read. This code mainly removed most of the `is_parent_create` checks and inlined everything in separate functions.

It made it easier to only do the idempotency check on nested connect inside update.

Reply:
It's easier to read than before, so 👍🏾